src/parquet/column/reader.cc

```diff
@@ -33,7 +33,10 @@ ColumnReader::ColumnReader(const ColumnDescriptor* descr,
   : descr_(descr),
     pager_(std::move(pager)),
     num_buffered_values_(0),
-    num_decoded_values_(0) {}
+    num_decoded_values_(0) {
+      repetition_level_decoder_.Init(descr->max_repetition_level());
+      definition_level_decoder_.Init(descr->max_definition_level());
```

Comment:
Right, what I was saying is that the first time Reset is called it would have to create the right decoder instance if it is null (i.e. on the first invocation). When the ColumnReader ctor is called the level encoding type is not known.